controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1309,7 +1309,7 @@ func clearHostProvisioningSettings(host *metal3api.BareMetalHost) {
 }
 
 func (r *BareMetalHostReconciler) actionDeprovisioning(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
-	if info.host.Status.Provisioning.Image.URL != "" {
+	if info.host.Status.Provisioning.Image.URL != "" || info.host.Status.Provisioning.CustomDeploy != nil {
```

Comment:
@dtantsur I think we also have to make a change to https://github.com/metal3-io/baremetal-operator/blob/af93711c586ed708e26217cf7a74caa1826a456e/pkg/provisioner/ironic/ironic.go#L1106-L1109 for this to have any effect, but in the case of CustomDeploy the `instance_info` is likely to be empty either way. I presume the place where it is getting cleared is https://opendev.org/openstack/ironic/src/branch/stable/2024.2/ironic/conductor/manager.py#L1149. There doesn't seem to be any corresponding code in ironic to clear the `deploy_interface`, so I don't know how we can distinguish being in the `manageable` state after a previous failure to clean from being in the `manageable` state after recreating the node in a fresh database. Do you have any ideas?

Reply:
On closer inspection, that code wasn't doing anything anyway, so I removed it.